From 1a0cc8ca8606a63a45debbb04915aeb07f9cbea2 Mon Sep 17 00:00:00 2001
From: Andreas Schmidt <andreas.schmidt@wago.com>
Date: Wed, 5 Sep 2018 13:57:30 +0200
Subject: [PATCH 239/261] net: cpsw: fix access to already freed skb

Increment of skb->dev->rx_dropped should be done before
call of kfree_skb(skb);

Signed-off-by: Andreas Schmidt <andreas.schmidt@wago.com>
---
 drivers/net/ethernet/ti/cpsw_async_net.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/drivers/net/ethernet/ti/cpsw_async_net.c b/drivers/net/ethernet/ti/cpsw_async_net.c
index ac348e7..dd4b5e0 100644
--- a/drivers/net/ethernet/ti/cpsw_async_net.c
+++ b/drivers/net/ethernet/ti/cpsw_async_net.c
@@ -84,9 +84,9 @@ int cpsw_queue_net_rx(struct sk_buff *skb)
 	}
 
 	if (drop_packet || !rx_work) {
+		atomic_long_inc(&skb->dev->rx_dropped);
 		kfree_skb(skb);
 		kfree(rx_work);
-		atomic_long_inc(&skb->dev->rx_dropped);
 		return NET_RX_DROP;
 	}
 
-- 
2.7.4

